experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+      for i in range(3):
+        make_heatmap(act=i, reward_fn=norm_rew_fn, gen_trajs=gen_trajs,
+                     exp_trajs=exp_trajs)
+        save_path = Path(output_dir, f"act_{i}", checkpoint_dir.name)
+        os.makedirs(save_path.parent, exist_ok=True)
+        plt.savefig(save_path)
+
+        make_heatmap(act=i, reward_fn=norm_rew_fn)
+        save_path = Path(output_dir, f"no_rollout_act_{i}",
+                         checkpoint_dir.name)
+        os.makedirs(save_path.parent, exist_ok=True)
+        plt.savefig(save_path)
+        print(f"saved to {save_path}")
+
+
+def plot_reward_vs_time(
+    trajs_dict: Dict[str, List[util.rollout.Trajectory]],
+    reward_fn: RewardFn,
+    colors: Sequence[str] = ("tab:blue", "tab:orange"),
+):
+  """Plots rewards received by many trajectories from many agents over time.
+
+  Args:
+      trajs_dict: Dictionary mapping rollout labels (e.g. "expert" or
+        "gen policy") to rollouts associated with those labels.
+      reward_fn: Reward function for evaluating rollout rewards.
+      colors: Custom colors for plotted rewards.
+  """
+  plt.clf()
+  for i, (traj_name, traj_list) in enumerate(trajs_dict.items()):
+    X = []
+    Y = []
+    for traj in traj_list:
+      T = len(traj.rews)
+      X.extend(range(T))
+      rews = reward_fn(traj.obs[:-1], traj.acts, traj.obs[1:], None)
+      Y.extend(rews)
+    if i < len(colors):
+      color = colors[i]
+    else:
+      color = None
+    plt.plot(X, Y, alpha=1.0, c=color, label=traj_name)
+  plt.xlabel("timestep")
+  plt.ylabel("test reward")
+  plt.legend()
+  plt.show()
+
+
+def _reward_fn_normalize_inputs(obs: np.ndarray,
+                                acts: np.ndarray,
+                                next_obs: np.ndarray,
+                                steps: Optional[np.ndarray] = None,
+                                *,
+                                reward_fn: RewardFn,
+                                vec_normalize: VecNormalize,
+                                norm_reward: bool = True,
+                                verbose: bool = True,
+                                ) -> np.ndarray:
+  """Combine with `partial` to create an input-normalizing RewardFn.
+
+  Args:
+    reward_fn: The reward function that normalized inputs are evaluated on.
+    vec_normalize: Instance of VecNormalize used to normalize inputs and
+     rewards.
+    norm_reward: If True, then also normalize reward before returning.
+  Returns:
+    The possibly normalized reward.
+  """
+  norm_obs = vec_normalize.norm_obs(obs)
+  norm_next_obs = vec_normalize.norm_obs(next_obs)
+  rew = reward_fn(norm_obs, acts, norm_next_obs, steps)
+  if norm_reward:
+    rew = vec_normalize.normalize_reward(rew)
+  if verbose:
+    print("rew normed min:", rew.min())
+    print("rew normed max:", rew.max())
+  return rew
+
+
+def build_norm_reward_fn(*, reward_fn, vec_normalize, **kwargs) -> RewardFn:
```

Comment:
Also it should be moved earlier in the code. Imagine if the code is a DAG, we want the source file to be a topological sort. (In practice there can be cycles which break this.)